server/sonar-server/src/main/java/org/sonar/server/setting/ws/SettingValidations.java

```diff
@@ -103,6 +91,41 @@ public SettingData(String key, List<String> values, @Nullable ComponentDto compo
       this.values = requireNonNull(values);
       this.component = component;
     }
+  }
+
+  private class ValueTypeValidation implements SettingValidation {
+
+    @Override
+    public void validate(SettingData data) {
+      PropertyDefinition definition = definitions.get(data.key);
+      if (definition == null) {
+        return;
+      }
+
+      if (definition.type() == PropertyType.METRIC) {
+        metric(data);
+      } else {
+        otherTypes(data, definition);
+      }
+    }
 
+    private void otherTypes(SettingData data, PropertyDefinition definition) {
+      data.values.stream()
+        .map(definition::validate)
+        .filter(result -> !result.isValid())
+        .findAny()
+        .ifPresent(result -> {
+          throw new BadRequestException(i18n.message(Locale.ENGLISH, "property.error." + result.getErrorKey(),
+            format("Error when validating setting with key '%s' and value [%s]", data.key, data.values.stream().collect(Collectors.joining(", ")))));
+        });
+    }
+
+    private void metric(SettingData data) {
+      try (DbSession dbSession = dbClient.openSession(false)) {
+        List<MetricDto> metrics = dbClient.metricDao().selectByKeys(dbSession, data.values);
```

Comment:
Please filter out disable metrics, it should not be allow to set a disable metric, whereas it's possible as it is now.